partiql-eval/src/test/kotlin/org/partiql/eval/CatalogTest.kt

```diff
@@ -0,0 +1,163 @@
+        val expected = Datum.bagVararg(
+            Datum.struct(
+                Field.of("a", Datum.doublePrecision(1.0)),
+                Field.of("b", Datum.doublePrecision(1.0)),
+                Field.of("C", Datum.doublePrecision(1.0))
+            ),
+            Datum.struct(
+                Field.of("a", Datum.doublePrecision(2.0)),
+                Field.of("b", Datum.doublePrecision(2.0)),
+                Field.of("C", Datum.doublePrecision(2.0))
+            )
+        )
+        assertEquals(0, Datum.comparator().compare(expected, datum))
+    }
+
+    @Test
+    fun absent_aggregate_extension_function() {
+        val query = "SELECT SUM(x) as a, foo_bar_baz(x) as b, FOO_BAR_BAZ(x) as C FROM << {'x':1}, {'x':2} >>"
+        val catalog = object : Catalog {
+            override fun getName(): String = "default"
+        }
+        val result = testQuery(query, catalog)
+        assert(result is TestResult.Failure)
+        val errs = (result as TestResult.Failure).err
+        assertTrue(errs.errors.isNotEmpty())
+        assertEquals(PError.FUNCTION_NOT_FOUND, errs.errors[0].code())
+    }
+
+    internal class AccumulatorSumDouble : org.partiql.spi.function.Accumulator {
+        var sum: Double = 0.0
+        var init = false
+
+        override fun next(args: Array<out Datum>?) {
+            val value = args!![0]
+            if (!init) {
+                init = true
+            }
+            val arg1 = value.double
+            sum += arg1
+        }
+
+        override fun value(): Datum {
+            return if (init) Datum.doublePrecision(sum) else Datum.nullValue(PType.doublePrecision())
+        }
+    }
+
+    @Test
+    fun aggregate_extension_function() {
+        val query = "SELECT SUM(x) as a, foo_bar_baz(x) as b, FOO_BAR_BAZ(x) as C FROM << {'x':1}, {'x':2} >>"
+        val catalog = object : Catalog {
+            override fun getName(): String = "default"
+            override fun getAggregations(session: Session, name: String): Collection<AggOverload> {
+                return listOf(
+                    // Just a `SUM(F64) -> F64` with a different name
+                    AggOverload.Builder("foo_bar_baz")
+                        .addParameters(PType.doublePrecision())
+                        .returns(PType.doublePrecision())
+                        .body { AccumulatorSumDouble() }
+                        .build()
+                )
+            }
+        }
+        val result = testQuery(query, catalog)
+        assert(result is TestResult.Success)
+        val datum = (result as TestResult.Success).result
+        val expected = Datum.bagVararg(
+            Datum.struct(
+                Field.of("a", Datum.doublePrecision(3.0)),
+                Field.of("b", Datum.doublePrecision(3.0)),
+                Field.of("C", Datum.doublePrecision(3.0))
+            ),
+        )
+        assertEquals(0, Datum.comparator().compare(expected, datum))
+    }
```

Comment:
wonder if we could also test a few other scenarios:
1. aggregate UDF has same name as a builtin scalar (e.g. UPPER)
2. scalar UDF has same name as a builtin aggregate (e.g. COUNT)
3. scalar UDF and aggregate UDF have same name